### PR TITLE
Create a Tekton Pipeline log generator.

### LINKED ISSUE #406: Create a Tekton Pipeline log generator.
We need an easy way to generate logs for development and testing.

We should create a CronJob that runs a Tekton `Pipeline` every minute to create job logs.

### PR DESCRIPTION
**Pull request by ggallen**
## Which Issue(s) this Pull Request Resolves

Resolves #406

## Release Note

```release-note
NONE
```

### CODE REVIEW

File: test/log-generators/pipelines/install.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Create a CronJob that run every minute, submitting a Tekton pipeline that generates an Apache
```
Comment (beerparty) [inline test/log-generators/pipelines/install.sh:3]: Small typo in the comment:

```suggestion
# Create a CronJob that runs every minute, submitting a Tekton pipeline that generates an Apache
```

File: test/log-generators/pipelines/install.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+# log file 1024 lines in length.
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```
Comment (beerparty) [inline test/log-generators/pipelines/install.sh:6]: What is the `SCRIPT_DIR` variable used for? I don't see it used anywhere in the script. If it is used in something called by this code, could you please add a comment with that information?

**Review by beerparty — CHANGES_REQUESTED** [review]

File: test/log-generators/pipelines/install.sh (hunk 2 above)
Comment (ggallen) [reply test/log-generators/pipelines/install.sh:6]: It's there because I expect @maruiz93 to request other changes that will need it!

Comment (beerparty) [reply test/log-generators/pipelines/install.sh:6]: If you end up not needing it, you should remove it. See also, [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

Comment (ggallen) [reply test/log-generators/pipelines/install.sh:6]: YAGNI!  Love it. :-)

File: test/log-generators/pipelines/install.sh [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+namespace="generate-logs-pipelines"
```
[the remark this answers is not in the document]
Comment (ggallen) [reply test/log-generators/pipelines/install.sh:8]: @maruiz93, I think your suggestion works for that first substitution.  But what about this one?
```
apiVersion: batch/v1
kind: CronJob
metadata:
  name: submit-generate-logs
spec:
  schedule: "*/1 * * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: submit
            image: curlimages/curl
            args: ["curl", "-X", "POST", "--data", "{}", "el-cron-listener.${namespace}.svc.cluster.local:8080"]
          restartPolicy: Never
```

**Review by rh-hemartin — APPROVED** [review]

File: test/log-generators/pipelines/install.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (maruiz93) [reply test/log-generators/pipelines/install.sh:8]: After trying out my own suggestion I think that using kustomize for this isn't doable.

First of all, the `vars` feature is deprecated in favor of `replacements` ([docs](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/)) that aren't that versatile.

Second, using `vars` won't work for resources that aren't listed [in here](https://github.com/kubernetes-sigs/kustomize/blob/a280cdf5eeb748f5a72c8d94164ffdd68d03c5ce/api/konfig/builtinpluginconsts/varreference.go) meaning.

So, the functionality is limited, what leads me to think that this isn't the right too for doing this replacement.

Then, I change my suggestion to just extract the YAML files and do a sed of the default namespace `generate-logs-pipelines` and apply it.

This makes the YAML applyable with a `kubectl apply -f` with the default values with the bonus of the highlighting

File: test/log-generators/pipelines/install.sh [hunk 4]
```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
```
Comment (mafh314) [inline test/log-generators/pipelines/install.sh:2]: Hey @ggallen could you add the license lines here, please?
`# Copyright KubeArchive Authors

# SPDX-License-Identifier: Apache-2.0
`

File: test/log-generators/pipelines/install.sh (hunk 1 above)
Comment (ggallen) [reply test/log-generators/pipelines/install.sh]: Fixed.

File: test/log-generators/pipelines/install.sh (hunk 4 above)
Comment (ggallen) [reply test/log-generators/pipelines/install.sh:4]: Fixed.!

File: test/log-generators/pipelines/install.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ggallen) [reply test/log-generators/pipelines/install.sh]: @maruiz93 , see my latest changes which use `envsubst` to do this.

[the remark this answers is not in the document]
Comment (maruiz93) [reply test/log-generators/pipelines/install.sh]: I like the way it looks now.

BTW, I see that you separated the k8s resources into it's own file. For me it's ok to have all the YAMLs together in the same file separated by `---`, specially if they are intended to be always applied together. The important thing for me was to not have mixed bash and YAML syntax in the same file :)

**Comment by mafh314** [comment]
@ggallen could you add some information about the log generator in the test [README.md](https://github.com/kubearchive/kubearchive/blob/main/test/README.md), please?

**Review by maruiz93 — APPROVED** [review]

File: test/log-generators/pipelines/install.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ggallen) [reply test/log-generators/pipelines/install.sh]: I separated them, and will leave them like that for now.

**Review by beerparty — APPROVED** [review]

**Pull request merged**